src/controller/userController.ts

```diff
@@ -218,8 +228,52 @@ export const Login = async (req: Request, res: Response) => {
   } catch (error) {
     res.status(500).send(error);
   }
+};
+
+export const updateProfile = async (req: Request, res: Response) => {
+  const userId: number = parseInt(req.params.id);
+  const { firstName, lastName, email, oldPassword, newPassword } = req.body as UpdateRrofileRequestBody;
+
+try {
+
+  const user = await userRepository.findOne({ where: { id: userId } });
+
+  if (!user) {
+    return res.status(404).json({ error: 'User not found' });
+  }
+  const passwordMatch = await bcrypt.compare(oldPassword, user.password);
+
+  if (!passwordMatch) {
+    return res.status(400).json({ error: 'Old password is incorrect' });
+  }
+  user.firstName = firstName || user.firstName;
+  user.lastName = lastName || user.lastName;
+  if (email && email !== user.email) {
+    const emailExists = await userRepository.findOne({ where: { email } });
+
+    if (emailExists) {
+      return res.status(400).json({ error: 'Email is already taken' });
+    }
+
+user.email = email;
+}
+if (newPassword) {
+  const hashedPassword = await bcrypt.hash(newPassword, 10); 
+  user.password = hashedPassword;
 }
+const errors = await validate(user);
 
+
+if (errors.length > 0) {
+  return res.status(400).json({ errors });
+}
```

Comment:
You have to validate your request body before anything else please, and please separate concerns, can you try to put validations aside to keep our controller clean